### PR TITLE
[PRM-650] Get previous uploads

### PR DESCRIPTION
**Pull request by AndrewRButcher**

### CODE REVIEW

File: app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html [hunk 1]
```diff
@@ -52,7 +52,7 @@ <h1 class="govuk-heading-l">@messages("otherReason.uploadEvidence.heading")</h1>
 
 
     @formHelper(action = postAction,
-        args = (Symbol("data-multi-file-upload-start-rows"), "1"), (Symbol("data-multi-file-upload-max-files"), "5"), (Symbol("id"), "multi-upload-form")) {
+        args = (Symbol("data-multi-file-upload-start-rows"), "1"), (Symbol("data-multi-file-upload-max-files"), "5"), (Symbol("id"), "multi-upload-form"), (Symbol("multiFileUploadUploadedFiles"), previousUploads)) {
```
Comment (LewisMitchell) [inline app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html:55]: `multiFileUploadUploadedFiles` should be `data-multi-file-upload-uploaded-files`, when the JS calls dataset on the form it gets parsed into camel-cased and drops the `data-` prefix

File: test/views/UploadEvidencePageSpec.scala [hunk 2]
```diff
@@ -48,7 +48,7 @@ class UploadEvidencePageSpec extends SpecBase with ViewBehaviours {
     val formProvider = UploadEvidenceForm.uploadEvidenceForm
 
     def applyView(form: Form[_], request: FakeRequest[_] = fakeRequest): HtmlFormat.Appendable = {
-      uploadEvidencePage.apply(form, controllers.routes.OtherReasonController.onSubmitForUploadEvidence(NormalMode))(request, implicitly, implicitly)
+      uploadEvidencePage.apply(form, controllers.routes.OtherReasonController.onSubmitForUploadEvidence(NormalMode), "")(request, implicitly, implicitly)
```
Comment (LewisMitchell) [inline test/views/UploadEvidencePageSpec.scala:51]: Is it possible to add a test to check that the form param has been filled in properly?

File: test/controllers/OtherReasonControllerSpec.scala [hunk 3]
```diff
@@ -46,12 +48,15 @@ class OtherReasonControllerSpec extends SpecBase {
       any(), any())
     ).thenReturn(authResult)
 
+    when(mockUploadJourneyRepository.getUploadsForJourney(any())).thenReturn(Future.successful(None))
```
Comment (LewisMitchell) [inline test/controllers/OtherReasonControllerSpec.scala:51]: Could we check the attribute in both the view and controller? For example, add a test here for when the repository returns some data?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]
Looks good, just a few things

File: app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html (hunk 1 above)
Comment (AndrewRButcher) [reply app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html]: Fixed

File: test/views/UploadEvidencePageSpec.scala (hunk 2 above)
Comment (AndrewRButcher) [reply test/views/UploadEvidencePageSpec.scala]: Added

File: test/controllers/OtherReasonControllerSpec.scala (hunk 3 above)
Comment (AndrewRButcher) [reply test/controllers/OtherReasonControllerSpec.scala]: Added

**Review by LewisMitchell — APPROVED** [review]

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**